pkg/queue/cluster_queue_interface.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	kueue "sigs.k8s.io/kueue/api/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+// ClusterQueue is an interface for a cluster queue to store workloads waiting
+// to be scheduled.
+type ClusterQueue interface {
+	Update(*kueue.ClusterQueue)
```

Comment:
Can you document the expectations for each function?

Reply:
added the document :)